### PR TITLE
PHO-1083 - Expose headers in rendering components

### PR DESCRIPTION
**Pull request by justinelam**
@matteofigus

### CODE REVIEW

**Comment by justinelam** [comment]
@matteofigus Yep, that makes sense! I'll work on those changes, the test and updating the README.md.

File: src/components-renderer.js [hunk 1]
```diff
@@ -39,22 +39,23 @@ module.exports = function(config, renderTemplate, templatesModules) {
...
           let hasErrors = false;
 
           _.each(toDo, action => {
             if (action.result.error) {
               hasErrors = true;
             }
-
             errors.push(action.result.error || null);
             results.push(action.result.html);
+            headers.push(action.result.headers);
```
Comment (richardhopton) [inline src/components-renderer.js]: This could be details.push({action.result.headers})

File: src/components-renderer.js [hunk 2]
```diff
@@ -39,22 +39,23 @@ module.exports = function(config, renderTemplate, templatesModules) {
       renderComponents(toDo, options, () => {
         processClientReponses(toDo, options, () => {
           const errors = [],
-            results = [];
+            results = [],
+            headers = [];
```
Comment (richardhopton) [inline src/components-renderer.js]: headers -> details

File: src/components-renderer.js [hunk 3]
```diff
@@ -39,22 +39,23 @@ module.exports = function(config, renderTemplate, templatesModules) {
...
             }
-
             errors.push(action.result.error || null);
             results.push(action.result.html);
+            headers.push(action.result.headers);
           });
 
           if (hasErrors) {
-            callback(errors, results);
+            callback(errors, results, headers);
```
Comment (richardhopton) [inline src/components-renderer.js]: headers -> details

File: src/components-renderer.js [hunk 4]
```diff
@@ -39,22 +39,23 @@ module.exports = function(config, renderTemplate, templatesModules) {
...
           } else {
-            callback(null, results);
+            callback(null, results, headers);
```
Comment (richardhopton) [inline src/components-renderer.js]: headers -> details

File: src/components-renderer.js (hunk 1 above)
Comment (richardhopton) [reply src/components-renderer.js]: (for now)

**Pull request merged**